source/simulator/core.cc

```diff
@@ -2422,11 +2421,11 @@ namespace aspect
         // if so, then loop over as many times as this is necessary
         if ((parameters.additional_refinement_times.size() > 0)
             &&
-            (parameters.additional_refinement_times.front () < time+time_step))
+            (parameters.additional_refinement_times.front () < time+new_time_step))
           {
             while ((parameters.additional_refinement_times.size() > 0)
                    &&
-                   (parameters.additional_refinement_times.front () < time+time_step))
+                   (parameters.additional_refinement_times.front () < time+new_time_step))
```

Comment:
Yes we should, your change is correct as well. The reason behind this is that we have already solved the timestep above, and here we are checking, whether the *next* timestep will be after an additional refinement time (everything that is happening here is preparing the variables for the next timestep).